librosa/core/spectrum.py

```diff
@@ -759,6 +761,197 @@ def perceptual_weighting(S, frequencies, **kwargs):
+    to scaling of the domain (e.g., time stretching or compression).  This is analogous
+    to the magnitude of the Fourier transform being invariant to shifts in the input domain.
+
+
+    .. [1] De Sena, Antonio, and Davide Rocchesso.
+        "A fast Mellin and scale transform."
+        EURASIP Journal on Applied Signal Processing 2007.1 (2007): 75-75.
+
+    .. [2] Cohen, L.
+        "The scale representation."
+        IEEE Transactions on Signal Processing 41, no. 12 (1993): 3275-3292.
+
+    Parameters
+    ----------
+    y : np.ndarray, real-valued
+        The input signal(s).  Can be multidimensional.
+        The target axis must contain at least 3 samples.
+
+    t_min : float > 0
+        The minimum time spacing (in samples)
+
+    n_fmt : int > 2 or None
+        The number of scale transform bins to use.
+        If None, then `n_bins = over_sample * ceil(n * log((n-1)/t_min))` is taken,
+        where `n = y.shape[axis]`
+
+    kind : str
+        The type of interpolation to use when re-sampling the input.
+        See ``scipy.interpolate.interp1d`` for possible values.
+
+    beta : float
+        The Mellin parameter.  beta=0.5 provides the scale transform.
+
+    over_sample : float >= 1
+        Over-sampling factor for exponential resampling.
+
+    axis : int
+        The axis along which to transform `y`
+
+    Returns
+    -------
+    x_scale : np.ndarray [dtype=complex]
+        The scale transform of `y` along the `axis` dimension.
+
+    Raises
+    ------
+    ParameterError
+        if `n_fmt < 2` or `t_min <= 0`
+        or if `y` is not finite
+        or if `y.shape[axis] < 3`.
+
+    Examples
+    --------
+    >>> # Generate a signal and time-stretch it (with energy normalization)
+    >>> scale = 1.5
+    >>> x1 = np.linspace(0, 1, num=512, endpoint=False)
+    >>> x2 = np.linspace(0, 1, num=scale * len(x1), endpoint=False)
+    >>> y1 = np.sin(2 * np.pi * x1)
+    >>> y2 = np.sin(2 * np.pi * x2) / np.sqrt(scale)
+    >>> # Verify that the two signals have the same energy
+    >>> np.sum(np.abs(y1)**2), np.sum(np.abs(y2)**2)
+        (256.0, 255.99999999999997)
+    >>> # We'll use cubic interpolation here for high precision
+    >>> scale1 = librosa.fmt(y1, n_fmt=200, kind='cubic')
+    >>> scale2 = librosa.fmt(y2, n_fmt=200, kind='cubic')
+    >>> # And plot the results
+    >>> import matplotlib.pyplot as plt
+    >>> plt.figure(figsize=(8, 4))
+    >>> plt.subplot(1, 2, 1)
+    >>> plt.plot(y1, label='Original')
+    >>> plt.plot(y2, label='Stretched')
+    >>> plt.xlabel('time (samples)')
+    >>> plt.title('Input signals')
+    >>> plt.legend(frameon=True)
+    >>> plt.axis('tight')
+    >>> plt.subplot(1, 2, 2)
+    >>> plt.semilogy(np.abs(scale1), label='Original')
+    >>> plt.semilogy(np.abs(scale2), label='Stretched')
+    >>> plt.xlabel('scale coefficients')
+    >>> plt.title('Scale transform magnitude')
+    >>> plt.legend(frameon=True)
+    >>> plt.axis('tight')
+    >>> plt.tight_layout()
```

Comment:
Does doing these calls here screw up doctests? I've had issues with that before. Not sure if you're running those in this project